File: src/main/java/com/kiafarhang/timely/batch/models/nyt/NYTStory.java
package com.kiafarhang.timely.batch.models.nyt;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.ZonedDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@JsonIgnoreProperties(ignoreUnknown = true)
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString(includeFieldNames = true)
public class NYTStory {

  private String section;
  private String subsection;
  private String title;

  @JsonProperty("abstract")
  private String storyAbstract;

  private String url;
  private String byline;

  @JsonProperty("item_type")
  private String itemType;

  private String source;

  @JsonProperty("updated_date")
  private ZonedDateTime updatedDate;

  @JsonProperty("created_date")
  private ZonedDateTime createdDate;

  @JsonProperty("published_date")
  private ZonedDateTime publishedDate;

  @JsonProperty("first_published_date")
  private ZonedDateTime firstPublishedDate;
}

File: src/main/java/com/kiafarhang/timely/batch/models/nyt/NYTStory.kt
package com.kiafarhang.timely.batch.models.nyt

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Builder
import java.time.ZonedDateTime

@JsonIgnoreProperties(ignoreUnknown = true)
@Builder
data class NYTStory(
        val section: String? = "",
        val subsection: String? = "",
        val title: String? = "",
        @JsonProperty("abstract") val storyAbstract: String? = "",
        val url: String? = "",
        val byline: String? = "",
        @JsonProperty("item_type") val itemType: String? = "",
        val source: String? = "",
        @JsonProperty("updated_date") val updatedDate: ZonedDateTime? = null,
        @JsonProperty("created_date") val createdDate: ZonedDateTime? =  null,
        @JsonProperty("published_date") val publishedDate: ZonedDateTime? = null,
        @JsonProperty("first_published_date") val firstPublishedDate: ZonedDateTime? = null
)